socket.go

```diff
@@ -232,8 +232,9 @@ func (socket *mongoSocket) InitialAcquire(serverInfo *mongoServerInfo, timeout t
 		stats.socketsExpired(+1)
 		debugf("socket: %p cannot use socket - max connection reuse time expired", socket)
 		socket.Unlock()
-		socket.Close()
-		return errors.New("Socket reuse time expired")
+		err := errors.New("Socket reuse time expired")
+		socket.kill(err, true)
```

Comment:
The only time mgo actually calls .Close on a socket is when mongoServer.Close, which does properly clears active sockets one by one. On all other places it uses kill(err, true), which properly removes socket from active and unused list.